appserver/ejb/ejb-rest-remoting/client/src/main/java/fish/payara/ejb/rest/client/RemoteEJBContext.java

```diff
@@ -0,0 +1,382 @@
+        if (environment.containsKey("fish.payara.readTimeout")) {
+            clientBuilder.readTimeout(getLong(environment.get("fish.payara.readTimeout")).longValue(), MICROSECONDS);
+        }
+        
+        if (environment.contains("fish.payara.hostnameVerifier")) {
+            clientBuilder.hostnameVerifier(getInstance(environment.get("fish.payara.hostnameVerifier"), HostnameVerifier.class));
+        }
+        
+        if (environment.contains("fish.payara.scheduledExecutorService")) {
+            clientBuilder.scheduledExecutorService(getInstance(environment.get("fish.payara.scheduledExecutorService"), ScheduledExecutorService.class));
+        }
+        
+        if (environment.contains("fish.payara.sslContext")) {
+            clientBuilder.sslContext(getInstance(environment.get("fish.payara.sslContext"), SSLContext.class));
+        }
+        
+        if (environment.contains("fish.payara.trustStore")) {
+            clientBuilder.trustStore(getInstance(environment.get("fish.payara.trustStore"), KeyStore.class));
+        }
+        
+        if (environment.contains("fish.payara.withConfig")) {
+            clientBuilder.withConfig(getInstance(environment.get("fish.payara.withConfig"), Configuration.class));
+        }
+        
+        return clientBuilder;
+    }
+    
+    private NamingException newNamingException(String name, Exception cause) {
+        NamingException namingException = new NamingException("Could not lookup :" + name);
+        namingException.initCause(cause);
+        
+        return namingException;
+    }
+    
+    private <T> T getInstance(Object value, Class<T> clazz) throws ClassNotFoundException, InstantiationException, IllegalAccessException {
+        if (clazz.isInstance(value)) {
+            return clazz.cast(value);
+        }
+        
+        if (value instanceof String) {
+            return clazz.cast(Class.forName((String)value).newInstance());
+        }
+        
+        throw new IllegalStateException("Value " + value + " has to be of type String or " + clazz);
+    }
+    
+    private Long getLong(Object value) {
+        if (value instanceof Number) {
+            return ((Number) value).longValue();
+        }
+        
+        if (value instanceof String) {
+            return Long.valueOf((String) value);
+        }
+        
+        throw new IllegalStateException("Value " + value + " has to be of type String or Number");
+    }
+    
+    private char[] getPassword(Object value) {
+        if (value instanceof String) {
+            return ((String) value).toCharArray();
+        }
+        
+        if (value instanceof char[]) {
+            return (char[]) value;
+        } 
+            throw new IllegalArgumentException("No password provided");
+        
+    }
+
+    @Override
+    public void bind(Name name, Object obj) throws NamingException {
+        
+    }
+
+    @Override
+    public void bind(String name, Object obj) throws NamingException {
+        
```

Comment:
I assume these are meant to be no-op and haven't just been forgotten?

Reply:
>I assume these are meant to be no-op and haven't just been forgotten?

Indeed, see class level comment:

> Note that at the moment only the <code>lookup</code> methods are implemented.